### PR TITLE
Updated sidebar display to show unapproved courses and interested students

### PR DESCRIPTION
**Pull request by bledsoef**
## Issue
Fixes issue #1146 and issue #1147 

We were wanting to the admin to be able to see important information like courses they need to approve and students who expressed interest since those require their action. 

## Changes
- Created a new controller in `admin/routes.py` called getSidebarInformation that gets all of the unapproved courses and all of the students who expressed interest in the CCE minor to display in the sidebar.
- Created a new JavaScript file `sidebar.js` that calls an ajax request on sidebar load and adds the numbers to the sidebar if applicable.
- Added hovers to each of the sidebar components that now have numbers next to them to display what the numbers mean.
- Modified `sidebar.html` to include the new script and gave the Course Management and Minor Management tabs an id to be referenced by the JavaScript script.

## Testing
- switch to branch `1146-interested-students-sidebar`
- use ramsayb2 and open up the admin dropdown.
- Hover over Course Management and Minor Management to make sure a popover appears and describes what the numbers mean.
- verify that Course Management has (1), that interested students has (4), and the Admin dropdown has (5).
- make a student express interest and a new course for the current term and make sure the numbers update when you return to ramsayb2.
- look over code.

### CODE REVIEW

File: app/static/js/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+$(document).ready(function() {
+    // fetch the number of interested students and display them  in the sidebar if there are any
```
Comment (zachneill) [inline app/static/js/sidebar.js]: there's an extra space here. dont you dare let this slide lawrence

**Review by zachneill — DISMISSED** [review]
urgent.

**Comment by andersoncedu** [comment]
Should the admin tab also show 'Admin (5)' ? This would let the admin know that there are items that need addressed in the admin tab.

**Comment by andersoncedu** [comment]
Additionally, I think it might be more noticeable if there was a color change when there was something requested or a box or something. Otherwise, the issue looks good.

**Comment by andersoncedu** [comment]
Without context, the numbers don't look like things that need to be addressed; they just look like numbers randomly on the page, in my opinion.

File: app/controllers/admin/minor.py [hunk 2]
```diff
@@ -8,10 +9,21 @@
...
                             sustainedEngagement = sustainedEngagement )
+
+@admin_bp.route('/admin/getInterestedStudentsCount', methods=['GET'])
+def getInterestedStudentsCount() -> str:
+    """
+    Get the count of students interested in the CCE minor to display in the 
+    admin sidebar. It must be returned as a string to be received by the
+    ajax request.
+    """
+    interestedStudentsCount: int = len(getMinorInterest())
```
Comment (stevensonmichel) [inline app/controllers/admin/minor.py]: I am not sure about the reason of int in that line, given that len always return an integer.

Comment (bledsoef) [reply app/controllers/admin/minor.py]: It is the type of the variable. I have just been trying to type my variables for better readability.

**Comment by bledsoef** [comment]
@andersoncedu In response to your comments, I should have commented this here but Brian and Anderson's suggestion was a hover to show more detail about the numbers. Otherwise since there are only a handful of admins and this will be demoed for them having it explicitly stated what the numbers are isn't super important.

**Comment by bledsoef** [comment]
[Let Admin see the amount of unapproved courses they have in the sidebar](https://trello.com/c/IKOp7eSs/737-let-admin-see-the-amount-of-unapproved-courses-they-have-in-the-sidebar)

**Comment by bledsoef** [comment]
[Display the amount of interested minor students in the sidebar](https://trello.com/c/Vb0E4ikc/142-display-the-amount-of-interested-minor-students-in-the-sidebar)

**Review by Karina-Agliullova — APPROVED** [review]
Good work!

**Review by gahimbaref — APPROVED** [review]
Looks good to me:)

**Comment by gahimbaref** [comment]
> Additionally, I think it might be more noticeable if there was a color change when there was something requested or a box or something. Otherwise, the issue looks good.

I think a color might not be necessary. Having the numbers there seems to be sufficient

**Pull request merged**